tests/test_pickle.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+import pickle
+
+import awkward as ak
+
+
+def test_pickle_ak_array():
+    buffers = []
+    array = ak.Array([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]])[[0, 2]]
+    next_array = pickle.loads(
+        pickle.dumps(array, protocol=5, buffer_callback=buffers.append), buffers=buffers
```

Comment:
This is a nice concise way to test pickle 5!
Can we add a line that confirms that `buffer` did in fact get some content?

That's all I have, otherwise good to go.